app/vmctl/vm_native.go

```diff
@@ -16,8 +16,9 @@ type vmNativeProcessor struct {
 	filter    filter
 	rateLimit int64
 
-	dst *vmNativeClient
-	src *vmNativeClient
+	dst     *vmNativeClient
+	src     *vmNativeClient
+	syncErr chan error
```

Comment:
wdyt about passing a context instead of channel?
For example:
```
 package main

@@ -26,6 +27,8 @@ func main() {
 
+       ctx, cancelCtx := context.WithCancel(context.Background())



-                                       return p.run()
+                                       return p.run(ctx)
                                },
                        },
                        {
@@ -214,6 +217,7 @@ func main() {
                if importer != nil {
                        importer.Close()
                }
+               cancelCtx()
        }()
```

and 
```
+func (p *vmNativeProcessor) run(ctx context.Context) error {
+               req = req.WithContext(ctx)

```